code/modules/ninja/energy_katana.dm

```diff
@@ -1,30 +1,87 @@
-/obj/item/weapon/katana/energy
+/obj/item/weapon/dash
+	name = "energy katana"
```

Comment:
things like the start, end visuals, the beam icon state + beam type, etc. should all be on this too.